GSA-TTS/FAC
# 2819 consolidate general information schema validation

## Description
This PR consolidates `GeneralInformationRequired.schema.jsonnet`, `GeneralInformationComplete.schema.jsonnet` and `GeneralInformation.schema.jsonnet`. The goal is to remove redundancy that could lead to confusion and potential bugs in the future.  Changes include:
- Merged `GeneralInformationComplete.schema.jsonnet` and `GeneralInformation.schema.jsonnet`
- Removed `GeneralInformationComplete.schema.jsonnet` 
- `GeneralInformationRequired.schema.jsonnet` enforces (i.e., makes mandatory) the following fields that were not enforced:
  `auditee_zip`,  `auditor_address_line_1`,  `auditor_city`,  `auditor_country`,  `audit_period_covered`.
- Improved validation error handling in general information form

## How to Test
- Launch the stack in local and switch to main branch. 
- Create a report and delete the period start date or end date before initiating pre-submission validation. 
- Trigger pre-submission validation and observe that the application crashes. Next, switch to this PR branch and re-run pre-submission validation and observe that the application does not crash.
- Because [3534](https://github.com/GSA-TTS/FAC/pull/3534) has been merged in, please test all the steps described in #3534 as well.

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
